pkg/gcp/client/compute.go

```diff
@@ -409,6 +413,34 @@ func (c *computeClient) ListRoutes(ctx context.Context, opts RouteListOpts) ([]*
 	return res, nil
 }
 
+func (c *computeClient) InsertAliasIPRoute(ctx context.Context, route apisgcp.Route, defaultSecondarySubnetName string) error {
+	log := shared.LogFromContext(ctx)
+
+	instance, err := c.GetInstance(ctx, route.Zone, route.InstanceName)
+	if err != nil {
+		return err
+	}
+
+	// Add an alias IP range to the first network interface
+	aliasIpRange := &compute.AliasIpRange{
+		IpCidrRange:         route.DestinationCIDR,
+		SubnetworkRangeName: defaultSecondarySubnetName,
+	}
+
+	// Append the alias IP range to the existing list
+	instance.NetworkInterfaces[0].AliasIpRanges = append(instance.NetworkInterfaces[0].AliasIpRanges, aliasIpRange)
+
+	var op *compute.Operation
+	op, err = c.service.Instances.UpdateNetworkInterface(c.projectID, route.Zone, route.InstanceName, instance.NetworkInterfaces[0].Name, instance.NetworkInterfaces[0]).Context(ctx).Do()
```

Comment:
You sure you don't want to wait for the operation to be completed. This currently is just a "promise" but you don't really wait for the async operation to complete before proceeding.